lib/puppet/ssl/host.rb

```diff
@@ -484,6 +479,28 @@ def load_certificate_bundle(bundle_string)
     end
   end
 
+  # Fetches the crl bundle from the CA server; because this is used for the first
+  # fetch of the crl, we need to pass in a certificate store that has no crls
+  # attached to it, but still utilizes the localcacert.
+  # @raise [Puppet::Error] if response from the server is not a valid certificate
+  #                        bundle
+  # @return nil
+  def download_and_save_crl_bundle
+    certificate_store = OpenSSL::X509::Store.new()
+    certificate_store.purpose = OpenSSL::X509::PURPOSE_ANY
+    certificate_store.add_file(Puppet.settings[:localcacert])
+
+    Puppet::Util.replace_file(crl_path, 0644) do |file|
+      Puppet::Rest::Routes.get_crls(http_client(ssl_store: certificate_store), CA_NAME) do |chunk|
+        file.write(chunk)
+      end
+    end
+    rescue Puppet::Error => e
```

Comment:
We should probably have an explict `begin` in this method to match the rest of the file.